test/src/test/integration/link_git_protocol.rs

```diff
@@ -341,6 +360,8 @@ where
         update_tips(&local_repo, &out.wanted_refs).unwrap();
     }
 
+    // Need to refresh it as it didn't notice the new pack
+    local_repo.refresh_object_database().unwrap();
```

Comment:
This is exactly what would have to be done on the server as well once it causes new packs to be written. Internally it currently replaces itself, which is certainly something that could be improved one day.